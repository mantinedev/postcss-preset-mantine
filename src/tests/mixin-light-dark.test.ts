import { testTransform } from './utils';

const baseInput = `
.demo {
  @mixin light {
    color: red;
  }

  @mixin dark {
    color: blue;
  }
}
`;

const whereInput = `
.demo {
  @mixin where-light {
    color: red;
  }

  @mixin where-dark {
    color: blue;
  }
}
`;

const rootInput = `
:root {
  @mixin light-root {
    color: red;
  }

  @mixin dark-root {
    color: blue;
  }
}
`;

const rootWhereInput = `
:root {
  @mixin where-light-root {
    color: red;
  }

  @mixin where-dark-root {
    color: blue;
  }
}
`;

describe('mixin-light-dark', () => {
  it('transforms light and dark mixins correctly', async () => {
    const res = await testTransform(baseInput);
    expect(res.css).toMatchSnapshot();
  });

  it('transforms where-light and where-dark mixins correctly', async () => {
    const res = await testTransform(whereInput);
    expect(res.css).toMatchSnapshot();
  });

  it('transforms light-root and dark-root mixins correctly', async () => {
    const res = await testTransform(rootInput);
    expect(res.css).toMatchSnapshot();
  });

  it('transforms where-light-root and where-dark-root mixins correctly', async () => {
    const res = await testTransform(rootWhereInput);
    expect(res.css).toMatchSnapshot();
  });
});
