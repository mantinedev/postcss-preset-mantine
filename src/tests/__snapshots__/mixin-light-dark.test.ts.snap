// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixin-light-dark transforms light and dark mixins correctly 1`] = `
"
[data-mantine-color-scheme='light'] .demo {
    color: red
}
  [data-mantine-color-scheme='dark'] .demo {
    color: blue
}
"
`;
